charts/datadog/templates/cluster-agent-deployment.yaml

```diff
@@ -178,14 +178,20 @@ spec:
             value: {{ .Values.clusterAgent.admissionController.mutateUnlabelled | quote }}
           - name: DD_ADMISSION_CONTROLLER_SERVICE_NAME
             value: {{ template "datadog.fullname" . }}-cluster-agent-admission-controller
-          {{- if .Values.clusterAgent.admissionController.configMode }}
           - name: DD_ADMISSION_CONTROLLER_INJECT_CONFIG_MODE
-            value: {{ .Values.clusterAgent.admissionController.configMode | quote }}
-          {{- if eq .Values.clusterAgent.admissionController.configMode "service" }}
+            {{- if .Values.clusterAgent.admissionController.configMode }}
+            value: {{ .Values.clusterAgent.admissionController.configMode }}
+            {{- else if or (eq .Values.datadog.apm.portEnabled true) (eq .Values.datadog.apm.enabled true) }}
+            value: service
```

Comment:
I'll need your help with this one! The Jira cards for this requested change mentioned this order of priority and I interpreted the config modes as: 

```
* Local Service Traffic if available  # service config mode
* TCP if hostPort Enable                  # service config mode
* else UDS                                         # socket config mode
```

Now that I'm looking at the Jira cards again I think it should be:

```
* Local Service Traffic if available  # service config mode if agents.localService.forceLocalServiceEnabled: true OR K8s 1.22+
* TCP if hostPort Enable                  # hostip config mode
* else UDS                                         # socket config mode
```

^ Is that correct? Let me know what you think

Reply:
Thanks for sharing more context!
Yea the second ordering makes more sense! Although I'm not 100% sure we should prioritize service over hostip. Historically hostip was the only mode and doesn't depend on dns nor on the k8s version. Users who want service can always configure it explicitly, that said, I'd follow this logic:
```
if configMode is not set
  if socket enabled, configMode == socket
  else if port enabled, configMode == hostip
```
The order matters, we want to chose the socket mode if both socket and port are enabled.

Any thoughts @yoannrobin ? Since you're familiar with this topic.